consensus/hotstuff/votecollector/statemachine.go

```diff
@@ -190,6 +190,11 @@ func (m *VoteCollector) ProcessBlock(proposal *model.Proposal) error {
 	}
 }
 
+// RegisterVoteConsumer effectively delegates call to votes cache
```

Comment:
lets add the full goDoc here to retain all the details from the interface specification:

```suggestion
// RegisterVoteConsumer registers a VoteConsumer. Upon registration, the collector
// feeds all cached votes into the consumer in the order they arrived.
// CAUTION, VoteConsumer implementations must be
//  * NON-BLOCKING and consume the votes without noteworthy delay, and
//  * CONCURRENCY SAFE
```